test/include/autoinstall/AutoinstallAskInclude_test.rb

```diff
@@ -0,0 +1,292 @@
+        }
+
+        it "creates a ComboBox widget" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
+          expected_options = [
+            client.Item(client.Id(:desktop), "Desktop", true),
+            client.Item(client.Id(:server), "Server", false)
+          ]
+          expect(client).to receive(:ComboBox).
+            with(client.Id("0_0"), client.Opt(:notify), ask["question"], expected_options).
+            and_call_original
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains a question with type 'boolean'" do
+        let(:ask) do
+          base_ask.merge("type" => "boolean", "question" => "Register system?", "default" => "true")
+        end
+
+        it "creates a CheckBox widget" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
+          expect(client).to receive(:CheckBox).
+            with(client.Id("0_0"), client.Opt(:notify), ask["question"], true).
+            and_call_original
+          client.askDialog
+        end
+      end
+    end
+
+    describe "dialogs actions" do
+
+      context "when ok button is pressed" do
+        let(:result) { :ok }
+        let(:value) { "some-value" }
+
+        before(:each) do
+          allow(Yast::UI).to receive(:QueryWidget).
+            with(client.Id("0_0"), :Value).and_return(value)
+        end
+
+        context "and a path was specified" do
+          let(:ask) { base_ask.merge("path" => "users,0,gecos") }
+
+          it "values are saved into the Profile at the specified path" do
+            expect(Yast::Profile).to receive(:setElementByList).
+              with(["users", 0, "gecos"], value, profile).
+              and_call_original
+            client.askDialog
+          end
+        end
+
+        context "and a pathlist was specified" do
+          let(:ask) { base_ask.merge("pathlist" => ["users,1000,login", "groups,1000,name"]) }
+
+          it "saves value into Profile at the paths specified in the pathlist" do
+            expect(Yast::Profile).to receive(:setElementByList).
+              with(["users", 1000, "login"], value, profile).and_call_original
+            expect(Yast::Profile).to receive(:setElementByList).
+              with(["groups", 1000, "name"], value, profile).and_call_original
+            client.askDialog
+          end
+        end
+
+        context "and a file was specified" do
+          let(:file_path) { "/tmp/value" }
+          let(:ask) { base_ask.merge("file" => file_path) }
+
+          it "saves value in the file" do
+            expect(Yast::SCR).to receive(:Write).
+              with(Yast::Path.new(".target.string"), file_path, value).
+              and_return(true)
+            client.askDialog
+          end
+
+          context "and value is a boolean" do
+            let(:value) { true }
+            let(:ask) { base_ask.merge("type" => "boolean", "file" => file_path)}
+
+            it "save the value in the file" do
```

Comment:
I think here we should be more specific what value is saved? how exactly serialized boolean looks like for true and for false.